enterprise_gateway/services/processproxies/crd.py

```diff
@@ -40,6 +42,53 @@ async def launch_process(
         await super().launch_process(kernel_cmd, **kwargs)
         return self
 
+    def get_container_status(self, iteration: int | None) -> str:
+        """Determines submitted CRD application status
+
+        Submitting a new kernel application to the spark operator
+        will take a while to be Running and can also fail before it
+        can actually start any pods. So we will validate that the
+        application succeeds on the Spark Operator side and then
+        delegate to parent to check if the application pod is running.
+
+        Returns
+        -------
+        Empty string if the container cannot be found otherwise.
+        The pod application status in case of success on Spark Operator side
+        Or the retrieved spark operator submission status in other cases (e.g. Failed)
+        """
+        with suppress(Exception):
+            custom_resource = client.CustomObjectsApi().get_namespaced_custom_object(
+                self.group,
+                self.version,
+                self.kernel_namespace,
+                self.plural,
+                self.kernel_resource_name,
+            )
+
+            if not custom_resource:
+                return ""
```

Comment:
I really don't like how the logs will "go silent" for about 10 seconds, where you would normally see status messages.  Would it be helpful to produce a message in the same format as what the superclass `get_container_status()` produces:
```python
        if iteration:  # only log if iteration is not None (otherwise poll() is too noisy)
            self.log.debug(
                f"{iteration}: Waiting to connect to k8s {self.object_kind.lower()} in "
                f"namespace '{self.kernel_namespace}'. Name: '{self.container_name}', "
                f"Status: '{pod_status}', Pod IP: '{self.assigned_ip}', KernelID: '{self.kernel_id}'"
            )
```
If you did nothing but address the variable names, it would look similar to this (which was produced by commenting out this method):
```
[D 2023-03-07 19:35:30.743 EnterpriseGatewayApp] 1: Waiting to connect to k8s sparkapplication in namespace 'jovyan-bb097194-63ed-4586-bdc1-0411279c48ff'. Name: '', Status: '', Pod IP: 'None', KernelID: 'bb097194-63ed-4586-bdc1-0411279c48ff'
```
We could probably drop "Pod IP" (and Status).